### PR TITLE
Add schema-stable and schema-nightly argoCD apps

### PR DESCRIPTION
**Pull request by iamvigneshwars**
Depends on #22

### CODE REVIEW

File: charts/apps/values.yaml [hunk 1]
```diff
@@ -8,6 +8,12 @@ source:
   graph:
     path: charts/graph
 
+schema:
+  repoURL: ghcr.io/diamondlightsource/graph-federation-supergraph
+  chart: supergraph
+  stableVersion: 0.1.1
+  nightlyVersion: 0.1.1
```
Comment (garryod) [inline charts/apps/values.yaml:15]: We should hardcode the `nightlyVersion` as `x` (i.e. whatever is newest)

File: charts/apps/values.yaml
Comment (garryod) [inline charts/apps/values.yaml:1]: Hmm, I didn't leave a particularly nice structure for you here, we should probably re-work it, perhaps:
- `monitoring`
- `graph`
- `schema`
Open to other suggestions though

File: charts/apps/templates/schema-nightly.yaml
Comment (garryod) [inline charts/apps/templates/schema-nightly.yaml:1]: Could we have this as a [second `source`](https://argo-cd.readthedocs.io/en/stable/user-guide/multiple_sources/) in the `graph-nightly` App?

File: charts/apps/templates/schema-stable.yaml
Comment (garryod) [inline charts/apps/templates/schema-stable.yaml:1]: Could we have this as a [second `source`](https://argo-cd.readthedocs.io/en/stable/user-guide/multiple_sources/) in the `graph-stable` App?

**Review by garryod — CHANGES_REQUESTED** [review]
Looking pretty good, even with the awkward values layout I left you with, should probably rework it a bit

File: charts/apps/values.yaml
Comment (iamvigneshwars) [reply charts/apps/values.yaml:1]: Yeah having them separated would look clean!

File: charts/apps/templates/schema-stable.yaml
Comment (iamvigneshwars) [reply charts/apps/templates/schema-stable.yaml:1]: Done

File: charts/apps/values.yaml (hunk 1 above)
Comment (iamvigneshwars) [reply charts/apps/values.yaml:15]: Got it

File: charts/apps/values.yaml
Comment (garryod) [reply charts/apps/values.yaml:1]: Nice one, that looks way better

File: charts/apps/templates/nightly.yaml [hunk 2]
```diff
@@ -7,42 +7,45 @@ metadata:
...
+                        endpoint: 'graph-monitoring-opentelemetry-collector:4317'
+                        protocol: grpc
+                      propagation:
+                        trace_context: true
+          oauth2-proxy:
+            ingress:
+              path: /nightly
+    - repoURL: {{ .Values.schema.repoURL }}
+      chart: {{ .Values.schema.chart }}
+      targetRevision: 0.1.1
```
Comment (garryod) [inline charts/apps/templates/nightly.yaml:48]: Surely this should be `x.x.x` or `x` or something?

File: charts/apps/Chart.yaml [hunk 3]
```diff
@@ -2,4 +2,4 @@ apiVersion: v2
 name: graph
 description: ArgoCD Apps used to deploy the Graph
 type: application
-version: 0.1.0
+version: 0.1.1
```
Comment (garryod) [inline charts/apps/Chart.yaml:5]: This feels like a breaking change given the structure of the `values.yaml` changes :sweat_smile:

File: charts/apps/values.yaml [hunk 4]
```diff
@@ -1,12 +1,19 @@
...
+  path: charts/monitoring
+
+graph:
+  repoURL: https://github.com/DiamondLightSource/graph-federation
+  targetRevision: HEAD
+  path: charts/graph
+
+schema:
+  repoURL: ghcr.io/diamondlightsource/graph-federation-supergraph
+  targetRevision: 0.1.1
```
Comment (garryod) [inline charts/apps/values.yaml:15]: In my mind we were going to use `x` and manually click "sync" in ArgoCD, but this is far more GitOps-y and keeps state in one place - I like it.

Perhaps we should turn on auto-sync in the stable chart and have the `graph` source point to a fixed release number too. What do you think?

**Review by garryod — CHANGES_REQUESTED** [review]
Looking good, I like the move to a proper GitOps style with the schema versioning, do you think we should extend that to the `graph` deployment too?

Also, what changes are needed to load the schema into the router? And can we get rid of the managed federation stuff?

File: charts/apps/values.yaml (hunk 4 above)
Comment (iamvigneshwars) [reply charts/apps/values.yaml:15]: Yeah, that's what I thought we were gna do, that's the reason I had targetVersion and stableVersion. we can make the graph point to the stableVersion and turn on auto-sync?

Comment (garryod) [reply charts/apps/values.yaml:15]: Wasn't what was in my head, but is probably the better route, I'd say go that way

File: charts/apps/templates/nightly.yaml [hunk 5]
```diff
@@ -7,42 +7,45 @@ metadata:
...
+                        endpoint: 'graph-monitoring-opentelemetry-collector:4317'
+                        protocol: grpc
+                      propagation:
+                        trace_context: true
+          oauth2-proxy:
+            ingress:
+              path: /nightly
+    - repoURL: {{ .Values.schema.repoURL }}
+      chart: {{ .Values.schema.chart }}
+      targetRevision: {{ .Values.schema.nightlyRevision }}
```
Comment (garryod) [inline charts/apps/templates/nightly.yaml:48]:
```suggestion
      targetRevision: "*"
```

File: charts/apps/templates/nightly.yaml [hunk 6]
```diff
@@ -7,42 +7,45 @@ metadata:
...
-  source:
-    repoURL: {{ .Values.source.repoURL }}
-    targetRevision: {{ .Values.source.targetRevision }}
-    path: {{ .Values.source.graph.path }}
-    helm:
-      valuesObject:
-        router:
+  sources:
+    - repoURL: {{ .Values.graph.repoURL }}
+      targetRevision: {{ .Values.graph.nightlyRevision }}
```
Comment (garryod) [inline charts/apps/templates/nightly.yaml:12]:
```suggestion
      targetRevision: HEAD
```

**Review by garryod — APPROVED** [review]

**Pull request merged**